torch/onnx/symbolic_opset9.py

```diff
@@ -1479,19 +1479,17 @@ def index_copy(g, self, dim, index, source):
 
 
 def type_as(g, self, other):
+    if sym_help._operator_export_type == torch.onnx.OperatorExportTypes.ONNX_ATEN_FALLBACK:
+        # We don't know the type of other, bail by emitting ATen
+        return g.op("ATen", self, other, operator_s="type_as")
+
     self_dtype = sym_help._try_get_scalar_type(self)
     other_dtype = sym_help._try_get_scalar_type(other)
     if self_dtype == other_dtype and self_dtype is not None:
         return self
-    if other_dtype is not None:
-        return g.op("Cast", self, to_i=sym_help.cast_pytorch_to_onnx[other_dtype])
-    else:
-        if sym_help._operator_export_type == torch.onnx.OperatorExportTypes.ONNX_ATEN_FALLBACK:
-            # We don't know the type of other, bail by emitting ATen
-            return g.op("ATen", self, other, operator_s="type_as")
-        else:
-            raise RuntimeError('Unsupported: ONNX export of type_as for tensor '
-                               'of unknown dtype.')
+    if other_dtype is None:
```

Comment:
This doesn't feel right, we cannot assume `Float` when type is not known. Do you see a case failing without doing this?